### ISSUE TITLE
Remove or make the out of date warning less in-your-face for patch releases

### ISSUE THREAD

**Issue by adamreisnz** [yarnpkg/yarn#3718]
**Do you want to request a *feature* or report a *bug*?**
Enhancement

**What is the current behavior?**
Yarn shows "warnings" about being out of date for patch releases, e.g.

```
yarn install v0.24.5
warning Your current version of Yarn is out of date. The latest version is "0.24.6"
```

**What is the expected behavior?**
The warning message in yellow is distracting, I really can't be bothered to install a new version of yarn every single time a new patch release comes out. And the yellow color always draws my attention and then the OCD kicks in :)

Unless there's a bug that's really bothering me, I am happy to upgrade yarn every now and then when I notice that I'm behind a few minor versions. In my opinion warnings should only show up if you're trailing a few minor versions, and maybe just show a muted/greyed out notice when it's just a patch or a single minor version.

**Comment by bestander** [comment]
Yeah, warnings suck if you see them a lot.
You can opt out with `disable-self-update-check` setting.

Curious to see more discussion here.

**Comment by adamreisnz** [comment]
Well removing it altogether is an option, but I think a warning can a good thing, provided it's actually warning you about something meaningful. Being one or two patch or even minor versions behind is not that big of a deal imo, unless one of them contains a critical/urgent fix. So maybe the version differences combined with some urgency flag could dictate when/how often warnings do show up.

**Comment by bestander** [comment]
From Yarn point of view we want everyone to be on the latest version, this is easier to triage bugs.
From developers point of view they want to be annoyed less and need to have meaningful and actionable warnings.

Can we make the warning less annoying while keeping people up to date?
The good thing is that this check only happens once a day.

Not sure if patch warnings should be a less priority though, we usually cherry-pick bug fixes worth the effort to do that so it sometimes is useful for people to be up to date.

**Comment by adamreisnz** [comment]
Fair enough, I understand the point of view from Yarn. Less annoying warnings (e.g. greyed out text) would help, with it turning yellow when you're behind a fair bit.

Maybe another crazy idea, but how far fetched would it be to have an opt-in auto update under the hood? Nearly every app and their dog out there are doing it, so why not look into that option?

**Comment by bestander** [comment]
We had it actually :)
One of the problems is that there are 10 ways to install Yarn across all OS, package managers etc and some of the require sudo permission.
When Yarn had autoupdater it would install another version of Yarn into a folder inside previous Yarn installation, this avoided the need to check for a permission.
But in the end it created a huge confusion regarding what is the current version of Yarn and which tool owns the update.
We will probably get back to this idea later with a better implementation but there are too many things to consider.

Feel free to send a PR with the different level of warning-ness depending on version difference.

**Comment by Daniel15** [comment]
The warning should only appear once per day, so it shouldn't be too annoying. We could change it to only show once per week instead maybe? Making it greyed out seems reasonable too.   

> Maybe another crazy idea, but how far fetched would it be to have an opt-in auto update under the hood?

We used to have a "self-update" command, so `yarn self-update` would update Yarn. However, it broke. #1139 tracks fixing it.

We could try auto update, however the update could take a while (which might not be what the user expects to happen when they run Yarn) and it may require root permissions.

**Comment by bestander** [comment]
```
yarn config set disable-self-update-check true
```
